src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs

```diff
@@ -261,7 +262,12 @@ public bool TryPersist(OptionKey optionKey, object value)
 
             var languagePreferences = new LANGPREFERENCES3[1];
             languagePreferences[0].guidLang = languageServiceGuid.Item1;
-            Marshal.ThrowExceptionForHR(_textManager.GetUserPreferences4(null, languagePreferences, null));
+
+            // shouldn't all these in SetUserPreferencesMayAsync?
+            var service = _serviceInitializer.GetService(CancellationToken.None);
+
+            // calling this here is safe when it is called from BG?
```

Comment:
I don't know. You'll need to confirm that the `SVsTextManager` service is not an STA object (and is thread-safe in addition to being free-threaded).

Reply:
the comment is question to @jasonmalinowski 

this API can called from BG, and textmanager is native COM STA service, so for writting, it does this trick to jump to UI thread, but not for read.